OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -285,13 +285,17 @@ public void onException(Exception ex) {
                             if (key.equalsIgnoreCase(fieldView.getField().getName())) {
                                 List<RegisterResponseFieldError> error = messageBody.get(key);
                                 showErrorOnField(error, fieldView);
-                                fieldErrorShown = true;
+                                if (!fieldErrorShown) {
```

Comment:
`fieldErrorShown` is logically not the right name here, cuz depending on this bool we are deciding to show the dialog, not the inline field error.

`errorDialogShown` sounds like a better name.